### PR TITLE
deprecated mixins @include H1  in SCSS replaced with Text component

### LINKED ISSUE #20496: Replace deprecated mixins `@include H1` - `@include H9` in SCSS with Text component
### Description

Currently, the extension is using outdated mixins `@include H1` - `@include H9`. This CSS should be removed and the JSX associated with these styles replaced with the `Text` component. This will reduce the amount of CSS in the extension and improve the cohesiveness of the UI.

This is a massive undertaking by itself and creating a single PR would be too large. Smaller PRs can be submitted against this issue to ensure easier review and gradual improvements.

### Technical Details

- Remove instances of the deprecated mixins from the CSS and replace the JSX element with `Text` component (`ui/components/component-library/text/text.tsx`) or [appropriate component from the component-library](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-introduction--docs).
- There will also be CSS that can be removed in favor of the style utility props that are available on the `Text` component. For example, `display: flex` can be replaced with `display={Display.Flex}` prop on the `Text` component. [Check out all the available style utility props in the Text docs in storybook](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-text--docs).
- The `Text` component has a `variant` prop that can be used to set the font size and weight. Use the `TextVariant` enum (`ui/helpers/constants/design-system.ts`) to set the appropriate variant.
- There are 9 deprecated mixins that need to be replaced. The following is a mapping of the mixin to the component that should be used to replace it:

`@include H1` => `<Text variant={TextVariant.displayMd}>`
`@include H2` => `<Text variant={TextVariant.displayLg}>`
`@include H3` => `<Text variant={TextVariant.headingMd}>`
`@include H4` => `<Text variant={TextVariant.headingSm}>`
`@include H5` => `<Text variant={TextVariant.bodyMd}>`
`@include H6` => `<Text variant={TextVariant.bodyMd}>`
`@include H7` => `<Text variant={TextVariant.bodySm}>`
`@include H8` => `<Text variant={TextVariant.bodyXs}>`
`@include H9` => `<Text variant={TextVariant.bodyXs}>`

### Acceptance Criteria

- Instances of deprecated mixins `@include H1` - `@include H9` are completely replaced with the `Text` component or appropriate component from the component-library
- Each Pull Request (PR) should include **no more than 3 files**
- The code changes should pass Jest tests, e2e tests, linting, and Storybook without any errors.
- The PR must include before and after screenshots of the UI to ensure there are no visual regressions.

If the acceptance criteria is not met, PRs may be closed.

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with running the extension locally, have knowledge of React, component props, Jest tests, linting, and Storybook, and want to contribute to improving the cohesiveness of UI in the extension

### PR DESCRIPTION
**Pull request by subhajit20**
## Explanation
Replaced deprecated mixin ` @include H1` with `<Text>` component 
Issue No - https://github.com/MetaMask/metamask-extension/issues/20496
Before:
before-2023-09-09 at 10 06 26 AM

After
Screenshot 2023-09-12 at 7 54 12 AM

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/pages/unlock-page/unlock-page.component.js [hunk 1]
```diff
@@ -176,7 +178,14 @@ export default class UnlockPage extends Component {
               </div>
             ) : null}
           </div>
-          <h1 className="unlock-page__title">{t('welcomeBack')}</h1>
+          <Text
+            as="h1"
+            variant={TextVariant.headingMd}
```
Comment (brad-decker) [inline ui/pages/unlock-page/unlock-page.component.js:183]: I believe that the mapping in the issue is h1 -> displayMd not headingMd and perhaps that's why in your screenshot there is a large difference in the size of the fonts

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: ui/pages/unlock-page/unlock-page.component.js (hunk 1 above)
Comment (subhajit20) [reply ui/pages/unlock-page/unlock-page.component.js:183]: Okay will do it in a while.
Thanks

Comment (subhajit20) [reply ui/pages/unlock-page/unlock-page.component.js]: resolved that. Can you check once?

**Comment by subhajit20** [comment]
@brad-decker can you check now?

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @subhajit20, thanks for your contribution. The code looks like a good start. I still need to do a visual review but it would be great to remove the unused CSS that is tied to the `unlock-page__title` https://github.com/MetaMask/metamask-extension/blob/f19e1f5f44b22aa7c1995b0cc3ac8b18ab1d3c71/ui/pages/unlock-page/index.scss#L37-L44

Also when I do a search for the `unlock-page__title` it returns some results that effect an e2e test this should be updated with a data-testid 
Screenshot 2023-09-13 at 8 45 08 AM

**Comment by subhajit20** [comment]
@georgewrmarshall  Hey can you tell me how to set `data-testId` to the `<Text/>` component ?

**Comment by subhajit20** [comment]
@brad-decker hey can you help?

**Comment by subhajit20** [comment]
> > @georgewrmarshall Is it similar to `className`?
> > Is it the right to set data-testId? `<Text as="h1" variant={TextVariant.displayMd} color={TextColor.textAlternative} marginTop={1} data-testid='unlock-page__title' > {t('welcomeBack')} </Text>`
> > and fetching the component? `const pageTitle = queryByTestId('unlock-page__title');`
> 
> Hey @subhajit20, use kebab-case for data-test ids and have a look at some existing e2e tests to find the format. Otherwise I'll try to come back to this today
> 
> ```
> data-testid="unlock-page-title"
> ```

Hey Done that.

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! I think there can be some more components updated on this page but because it's the first page user's see it is sensitive to change. Thanks for your contribution @subhajit20 w

**Comment by subhajit20** [comment]
@georgewrmarshall after this pr shall I make changes to other components or not?

**Comment by georgewrmarshall** [comment]
Hey @subhajit20, let me discuss it with the team and get back to you. It may require some collaboration with marketing e.g.  a tweet from our official account to update users of the UI changes. Would you be able to update the snapshots and possibly look into why the other e2e test is failing?

**Comment by georgewrmarshall** [comment]
Hey @subhajit20, could you please update the snapshots for this PR and we can get this in. You should be able to update snapshots using:
```
yarn jest ui/pages/unlock-page --updateSnapshot 
```

**Comment by subhajit20** [comment]
@georgewrmarshall  hey sorry for the delay, I am outside right now. As soon as I reach I will make the changes.

**Comment by subhajit20** [comment]
> Hey @subhajit20, could you please update the snapshots for this PR and we can get this in. You should be able to update snapshots using:
> 
> ```
> yarn jest ui/pages/unlock-page --updateSnapshot 
> ```

Done that.

**Review by georgewrmarshall — APPROVED** [review]
LGTM!

**Pull request closed without merging**